appserver/payara-appserver-modules/jaxrs-client-tracing/src/main/java/fish/payara/requesttracing/jaxrs/client/PayaraTracingServices.java

```diff
@@ -40,111 +40,117 @@
      */
-    public ServiceLocator getBasicServiceLocator() {
-        checkInitialized();
-        return basicServiceLocator;
-    }
+    public PayaraTracingServices() {
+        final ServiceLocator baseServiceLocator = Globals.getStaticBaseServiceLocator();
 
+        requestTracingService = getFromServiceHandle(baseServiceLocator, RequestTracingService.class);
+        openTracingService = getFromServiceHandle(baseServiceLocator, OpenTracingService.class);
+        invocationManager = getFromServiceHandle(baseServiceLocator, InvocationManager.class);
+    }
 
     /**
-     * @return {@link RequestTracingService}
+     * @return {@link RequestTracingService}, or null if the HK2 service couldn't be
+     *         initialised.
      */
     public RequestTracingService getRequestTracingService() {
-        checkInitialized();
-        return requestTracingService;
+        if (checkInitialized()) {
+            return requestTracingService;
+        }
+        return null;
     }
 
-
     /**
-     * @return {@link OpenTracingService}
+     * @return {@link OpenTracingService}, or null if the HK2 service couldn't be
+     *         initialised.
      */
     public OpenTracingService getOpenTracingService() {
-        checkInitialized();
-        return openTracingService;
+        if (checkInitialized()) {
+            return openTracingService;
+        }
+        return null;
     }
 
-
     /**
      * @return {@link InvocationManager}
      */
     public InvocationManager getInvocationManager() {
-        checkInitialized();
-        return basicServiceLocator.getService(InvocationManager.class);
+        if (checkInitialized()) {
+            return invocationManager;
+        }
+        return null;
     }
 
-
     /**
-     * @return application name known to the actual {@link InvocationManager}.
+     * @return application name known to the actual {@link InvocationManager}, or
+     *         null if no invocation manager can be found.
      */
     public String getApplicationName() {
-        final InvocationManager invocationManager = getInvocationManager();
-        if (invocationManager == null) {
-            return null;
+        if (checkInitialized()) {
+            return openTracingService.getApplicationName(invocationManager);
         }
-        final OpenTracingService otService = getOpenTracingService();
-        return otService == null ? null : otService.getApplicationName(invocationManager);
+        return null;
     }
 
-
     /**
-     * @return actually active {@link Tracer} for the current application.
+     * @return actually active {@link Tracer} for the current application, or null
+     *         if the tracing service is not available.
      */
     public Tracer getActiveTracer() {
         final String applicationName = getApplicationName();
-        if (applicationName == null) {
+        if (applicationName == null || !checkInitialized()) {
```

Comment:
No need to call the check - it will be null if applicationName is null and applicationName is null if check ...